Preserve deep links into Google docs

maxine opened the pull request:
### Description of Change
Check if a link to a Google document (docs.google.com) contains a deep link to a specific sub-section (heading). If so, append the deep link to the document url.

### Related Issue
#24 

### Motivation and Context
As @abstrctn (👋!) noted in his issue, a link to a specific heading in a Google doc is currently stripped from the link added to the Library page, so the user is directed to the Google doc generally instead of the specified heading in that document. 

### Checklist
<!-- Cross out items that don't apply and leave a short description of why the item isn't relevant. Check off ([x]) items you complete. -->

- [x] Ran `npm run lint` and updated code style accordingly
- [x] `npm run test` passes
- [X] PR has a description and all contributors/stakeholder are noted/cc'ed
~- [ ] tests are updated and/or added to cover new code~
~- [ ] relevant documentation is changed and/or added~

isaacwhite commented:
Hey @maxine, thanks for taking a stab at this!

I ran this locally and wasn't seeing the expected behavior. Instead I ended up with the original link, not the Library version. Looks like destructuring the result of `{} + ''` doesn't throw an error and instead just returns `undefined`.

On a stylistic note, would it be possible to avoid the nested ternary here? Maybe something like the following:

```
const [deepLink = ''] = decoded.match(/#heading([^/+])(.*)/i) || []

const { path: libraryPath } = isDoc ? list.getMeta(docId) : {}
// always include the deep link as part of the string; default will be empty
const libraryDeepLink = libraryPath && ``${libraryPath}${deepLink}`

$(el).attr('href', libraryDeepLink || decoded)
```

isaacwhite commented on server/formatter.js:
```diff
@@ -81,10 +81,13 @@ function normalizeHtml(html) {
+      const [deepLink = ''] = decoded.match(/(?<=heading([^/+]))(.*)/i) || []
 
-      const {path: libraryPath} = isDoc ? list.getMeta(docId) || {} : {}
+      const { path: libraryPath } = isDoc ? list.getMeta(docId) : {}
```
It looks like a change was introduced here that no longer provides the fallback if the docId does not match as expected. This was resulting in an uncaught error when I hit it. Can we put back the fallback here? Otherwise this looks great.

```suggestion
      const { path: libraryPath } = isDoc ? list.getMeta(docId) || {} : {}
```

maxine commented:
Got it -- thanks @isaacwhite !

isaacwhite commented:
@maxine last thing: I now see that this change appends a `#` to normal library links (that do not have a heading included). Could we avoid that?

maxine commented:
@isaacwhite good catch. I updated that and changed the regex slightly so that the lookbehind is separate from the capture: `(?<=#heading=)([^/]+)` instead of `(?<=#heading=([^/]+))`

isaacwhite commented:
Sorry for the delay on this, @maxine. This looks great. Thank you!
